concent_api/common/decorators.py

```diff
@@ -54,13 +54,38 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
     return wrapper
 
 
-def non_nesting_atomic(using: Union[str, Callable], savepoint: bool=True) -> Atomic:
-    if (
-        settings.DETECT_NESTED_TRANSACTIONS and
-        get_connection(using).in_atomic_block
-    ):
-        raise ConcentPendingTransactionError
-    if callable(using):
-        return Atomic(DEFAULT_DB_ALIAS, savepoint)(using)
-    else:
-        return Atomic(using, savepoint)
+class non_nesting_atomic(ContextDecorator):
+    """
+    This class acts as either decorator or context manager.
+    It discovers when the callable or code is being run in nested database transaction and raises exception.
+    Otherwise it wraps callable or code in database transaction.
+    """
+
+    def __init__(self, using: Union[str, Callable], savepoint: bool=True):
+        self.using = using
+        self.savepoint = savepoint
```

Comment:
this field is unnecessary, please remove it

Reply:
I would keep it, without it we won't have possibility to call `Atomic()` with this parameter if we need to. And there is no extra logic behind that.

Reply:
but you can still call `Atomic` in this way:
```python
self.atomic_context_decorator = Atomic(self.using, savepoint)
```

Reply:
:+1: done